### PR TITLE
Fix removal of WEF legacy manifest file

### PR DESCRIPTION
**Pull request by TCourtneyOwen**
- Also fix removal of manifest when selected from prompt
- Add tests to validate manifest removal
- Bunch of minor changes related to formatting util.ts

### CODE REVIEW

File: src/test/test.ts [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+            if (process.platform === "win32") {
+                manifests = await officeToolbox.getManifests(application);
+            } else {
+                testManifestPath = await officeToolbox.getSideloadManifestPath(testManifestPath, application);
+                manifests = await officeToolbox.getManifestsFromSideloadingDirectory(application);
+
+            }
+
+            manifests.forEach(element => {
+                if (element = testManifestPath) {
```
Comment (akrantz) [inline src/test/test.ts]: one = only seems incorrect here

Comment (akrantz) [reply src/test/test.ts]: should it validate the expected path?

File: src/test/test.ts [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+import * as path from "path";
+import * as util from "util";
+const application = "excel";
+
+describe("Add manifest", function () {
+    describe(process.platform === "win32" ? "Add manifest to the registry" : "Add the manifest to the WEF folder", function () {
+        this.timeout(0);
+        let manifestFound = false;
+        let manifests: string[];
+        let testManifestPath = path.resolve(`${process.cwd()}/src/test/test-manifest.xml`);
```
Comment (akrantz) [inline src/test/test.ts]: Should the "test" folder be moved so it is outside the "src/" folder?

File: src/test/test.ts [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+            if (process.platform === "win32") {
+                manifests = await officeToolbox.getManifests(application);
+            } else {
+                testManifestPath = await officeToolbox.getSideloadManifestPath(testManifestPath, application);
+                manifests = await officeToolbox.getManifestsFromSideloadingDirectory(application);
+
+            }
+
+            manifests.forEach(element => {
+                if (element = testManifestPath) {
```
Comment (akrantz) [inline src/test/test.ts]: single = here too

File: src/util.ts [hunk 4]
```diff
@@ -164,29 +166,45 @@ function getManifestsFromSideloadingDirectory(inputApplication: string): Promise
...
         if (!fs.existsSync(sideloadingDirectory)) {
           continue;
         }
 
-        const sideloadingManifestPath = await getSideloadManifestPath(manifestPathToRemove, application);
+        let sideloadingManifestPath = undefined;
+        let legacySideloadingManifestPath = undefined;
+        if (manifestSelected) {
+          sideloadingManifestPath = manifestPathToRemove;
+        }
```
Comment (akrantz) [inline src/util.ts]: formatting: `} else {` on one line

File: src/util.ts [hunk 5]
```diff
@@ -347,9 +365,17 @@ function validateManifest(manifestPath: string): Promise<string> {
   });
 }
 
-async function getSideloadManifestPath(manifestPath: string, application: string): Promise<string> {
+export function getSideloadingManifestDirectory(application: string): string {
```
Comment (akrantz) [inline src/util.ts:367]: nice to see you refactor this into a function that can be reused.

File: src/test/test.ts (hunk 1 above)
Comment (TCourtneyOwen) [reply src/test/test.ts]: I don't following the first comment about 'one =only seems incorrect'

Comment (akrantz) [reply src/test/test.ts]: one equal is assignment not test for equality.

File: src/test/test.ts (hunk 3 above)
Comment (TCourtneyOwen) [reply src/test/test.ts]: oh crap dumb mistake

File: src/test/test.ts (hunk 2 above)
Comment (TCourtneyOwen) [reply src/test/test.ts]: I was trying to do that initally and was hittinga ts-node error when running the tests: "TypeError: test/test.ts Emit skipped. I will do a bit more research on how to work around this error and will move the test folder under root if/when I have a fix

Comment (TCourtneyOwen) [reply src/test/test.ts]: Fixed issue by removing rootDir setting from tsconfig.json and moved test folder to root

**Review by akrantz — APPROVED** [review]

**Pull request merged**